File: Gestion_Evenement2/src/main/java/edu/esprit/entities/QuizScore.java
package edu.esprit.entities;

public class QuizScore {
    private int idScore;
    private int idQuiz;
    private int idUser;
    private float score;

    public QuizScore(){ idScore = 0;}
    public QuizScore(int idScore, int idQuiz, int idUser, float score) {
        this.idScore = idScore;
        this.idQuiz = idQuiz;
        this.idUser = idUser;
        this.score = score;
    }

    public int getIdScore() {
        return idScore;
    }

    public void setIdScore(int idScore) {
        this.idScore = idScore;
    }

    public int getIdQuiz() {
        return idQuiz;
    }

    public void setIdQuiz(int idQuiz) {
        this.idQuiz = idQuiz;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public float getScore() {
        return score;
    }

    public void setScore(float score) {
        this.score = score;
    }
}

File: projettest-main2/src/Entity/Quizscore.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Repository\QuizscoreRepository;

#[ORM\Entity(repositoryClass: QuizscoreRepository::class)]
class Quizscore
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idscore = null;

    #[ORM\ManyToOne(targetEntity:Quiz::class , inversedBy:' Quizscore')]
    private ?Quiz $idquiz;

    #[ORM\Column]
    private ?float $score ;

    #[ORM\ManyToOne(targetEntity:Utilisateurs::class , inversedBy:' Quizscore')]
    private ?Utilisateurs $iduser;

    public function getIdscore(): ?int
    {
        return $this->idscore;
    }

    public function getIdquiz(): ?int
    {
        return $this->idquiz;
    }

    public function setIdquiz(int $idquiz): static
    {
        $this->idquiz = $idquiz;

        return $this;
    }

    public function getScore(): ?float
    {
        return $this->score;
    }

    public function setScore(float $score): static
    {
        $this->score = $score;

        return $this;
    }

   /* public function getIduser(): ?Utilisateurs
    {
        return $this->iduser;
    }*/

    public function setIduser(?Utilisateurs $iduser): static
    {
        $this->iduser = $iduser;

        return $this;
    }


}